### DISCUSSION TITLE
Material Shader affects each particle individually 4.0.

### DISCUSSION THREAD

**Discussion by Zerostin** [godotengine/godot-proposals#6667] [2D]
I've been tinkering around with VFX trying to establish a system for how I approach them in my game. One approach I've tried is to use the alpha of a texture as a way to clamp the colors of a particle system. The only problem with this approach is that the shader makes a pass on each particle independently, not as a whole. The current blend modes don't really provide a good solution to this so I was wondering if anyone here might know a way to make a shader affect an entire particle system as a whole.

**Comment by Calinou** [comment]
I don't see a way to do this in 3D, but in 2D, you can use the CanvasGroup node and use a shader on the CanvasGroup node itself.
